### ISSUE TITLE
CSP Error and Jquery

### ISSUE THREAD

**Issue by TomJahncke** [jquery/jquery#3117]
With CSP enabled in an web application and *not* allowing "unsafe-line" styles the jquery library encounter and error which goes to the browser console.  The jquery library does handle have a fallback method when innerHTML is not possible, so the application is functional, just lots of false positives in the console.  My request is to have the library proactively check if it is allow to use innerHTML rather than just relying on the exception path to address this.  

I am encountering this in Chrome on a Windows 7 machine.  

Here is the content security policy configuration that results in the this warning.  

"Content-Security-Policy","default-src 'self'; style-src 'self'"

jquery.js:5513 Refused to apply inline style because it violates the following Content Security Policy directive: "style-src 'self'". Either the 'unsafe-inline' keyword, a hash ('sha256-IOWYPccv4+GIAWz50PQ4hgBzwty+G8ckj9XrN5jdx6g='), or a nonce ('nonce-...') is required to enable inline execution.

In Jquery 2.2.2 the line is 5513 and the content looks like:

```js
			// See if we can take a shortcut and just use innerHTML
			if ( typeof value === "string" && !rnoInnerhtml.test( value ) &&
				!wrapMap[ ( rtagName.exec( value ) || [ "", "" ] )[ 1 ].toLowerCase() ] ) {

				value = jQuery.htmlPrefilter( value );

				try {
					for ( ; i < l; i++ ) {
						elem = this[ i ] || {};

						// Remove element nodes and prevent memory leaks
						if ( elem.nodeType === 1 ) {
							jQuery.cleanData( getAll( elem, false ) );
							elem.innerHTML = value;
						}
					}

					elem = 0;

				// If using innerHTML throws an exception, use the fallback method
				} catch ( e ) {}
			}

```

EDIT (by @mgol): Fixed Markdown code formatting so that it's readable.

** Description**

** Link to test case **

**Comment by mgol** [comment]
How would you check if `innerHTML` is allowed, though? If we did an initial `try-catch` to check, we'd always log a warning once to the console.

**Comment by dmethvin** [comment]
@TomJahncke can you provide a link to a test case? I'm guessing your HTML has some inline styles but usually it's better not to guess and that's why we ask for a test case as indicated by the ** Link to test case ** that was in the issue template.

**Comment by TomJahncke** [comment]
Thank you for your quick replies.  @dmethvin in attempting to provide a test case I did find a few places in my HTML where I was using a style instead of a class, those have been addressed.  Now, the only remaining times I get the CSP warning are related to an independent angular library we are using.  I am going to reach out to the angular library for assistance (I was able to create a simple example to demonstrate the issue with this library).   

I am content with closing this issue as there is nothing for jquery to do.  

Once again thank you for quick reply and patience me submitting this issue.
